tests/phpunit/integration/Core/Authentication/Google_ProxyTest.php

```diff
@@ -43,6 +44,134 @@ public function test_get_site_fields() {
 		);
 	}
 
+	public function test_fetch_site_fields() {
+		$context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$google_proxy = new Google_Proxy( $context );
+		$credentials  = new Credentials( new Options( $context ) );
+
+		$this->fake_proxy_site_connection();
+
+		$pre_args = null;
+		$pre_url  = null;
+
+		// Use pre_http_request for backwards compatibility as http_api_debug is not fired for blocked requests before WP 5.3
+		add_filter(
+			'pre_http_request',
+			function ( $false, $args, $url ) use ( &$pre_args, &$pre_url ) {
+				$pre_args = $args;
+				$pre_url  = $url;
+
+				return $false;
+			},
+			10,
+			3
+		);
+
+		// Mock reponse.
+		$mock_response = array(
+			'site_id',
+			'site_secret',
+			'url',
+			'name',
+			'redirect_uri',
+			'return_uri',
+			'action_uri',
+			'analytics_redirect_uri',
+		);
+		$this->mock_http_request(
+			$google_proxy->url( Google_Proxy::OAUTH2_SITE_URI ),
+			$mock_response
+		);
+		$google_proxy->fetch_site_fields( $credentials );
+
+		// Ensure the request was made with the proper URL and body parameters.
+		$this->assertEquals( $google_proxy->url( Google_Proxy::OAUTH2_SITE_URI ), $pre_url );
+		$this->assertEquals( 'POST', $pre_args['method'] );
+
+		$this->assertEqualSets(
+			array(
+				'site_id',
+				'site_secret',
+			),
+			array_keys( $pre_args['body'] )
+		);
+	}
+
+	public function test_are_site_fields_synced() {
+		$context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$google_proxy = new Google_Proxy( $context );
+		$credentials  = new Credentials( new Options( $context ) );
+
+		$fake_creds = $this->fake_proxy_site_connection();
+
+		$pre_args = null;
+		$pre_url  = null;
+
+		add_filter(
+			'pre_http_request',
+			function ( $false, $args, $url ) use ( &$pre_args, &$pre_url ) {
+				$pre_args = $args;
+				$pre_url  = $url;
+
+				return $false;
+			},
+			10,
+			3
+		);
+
+		// Mock matching reponse.
+		$matching_mock_response = array(
+			'url'                    => home_url(),
+			'action_uri'             => admin_url( 'index.php' ),
+			'name'                   => get_bloginfo( 'name' ),
+			'return_uri'             => $context->admin_url( 'splash' ),
+			'redirect_uri'           => add_query_arg( 'oauth2callback', 1, admin_url( 'index.php' ) ),
+			'analytics_redirect_uri' => add_query_arg( 'gatoscallback', 1, admin_url( 'index.php' ) ),
+		);
+		$this->mock_http_request(
+			$google_proxy->url( Google_Proxy::OAUTH2_SITE_URI ),
+			$matching_mock_response
+		);
+		$success_response_data = $google_proxy->are_site_fields_synced( $credentials );
+
+		// Ensure the request was made with the proper URL and body parameters.
+		$this->assertEquals( $google_proxy->url( Google_Proxy::OAUTH2_SITE_URI ), $pre_url );
+		$this->assertEquals( 'POST', $pre_args['method'] );
+		$this->assertEqualSetsWithIndex(
+			array(
+				'site_id',
+				'site_secret',
+			),
+			array_keys( $pre_args['body'] )
+		);
```

Comment:
@benbowler no need to assert functionality from the `fetch_site_fields` method. It is already checked in its own test.